visualization/app/codeCharta/ui/scenarioDropDown/scenarioDropDown.component.ts

```diff
@@ -3,34 +3,124 @@
 import "./scenarioDropDown.component.scss"
 import { ScenarioHelper, Scenario } from "../../util/scenarioHelper"
 import { MetricService, MetricServiceSubscriber } from "../../state/metric.service"
-import { MetricData } from "../../codeCharta.model"
+import { ColorRange, MetricData } from "../../codeCharta.model"
 import { IRootScopeService } from "angular"
-import { ThreeOrbitControlsService } from "../codeMap/threeViewer/threeOrbitControlsService"
 import { StoreService } from "../../state/store.service"
 import { setState } from "../../state/store/state.actions"
+import { DialogService } from "../dialog/dialog.service"
+import { setColorRange } from "../../state/store/dynamicSettings/colorRange/colorRange.actions"
+import { ThreeOrbitControlsService } from "../codeMap/threeViewer/threeOrbitControlsService"
 
 export class ScenarioDropDownController implements MetricServiceSubscriber {
 	private _viewModel: {
 		scenarios: Scenario[]
```

Comment:
I think it might make sense to not use the exact scenario array inside the viewModel, but a separate version, which also includes the `fa-icon` strings. Then you can just loop over the array inside the html template.